src/circuits/gates.jl

```diff
@@ -409,30 +437,39 @@ gate(s::String, args...; kwargs...) = gate(GateName(s), args...; kwargs...)
 # Version that accepts a dimension for the gate,
 gate(gn::GateName, dims::Tuple; kwargs...) = gate(gn; kwargs...)
 
-
 """
     combinegates(gn::GateName, s::Tuple; kwargs...)
 
 Generate a gate (matrix) given a set of operations in the gatename string
 """
 function combinegates(gn::GateName, s::Tuple; kwargs...)
-  name = string(ITensors.name(gn))
-  name = filter(x -> !isspace(x), name)
+  name = ""
+  gate1 = nothing
+  gate2 = nothing
+
+  ignore() do 
+    name = string(ITensors.name(gn))
+    name = filter(x -> !isspace(x), name)
+  end
   
   # first check for addition
   pluspos = findfirst("+", name)
   if !isnothing(pluspos)
     !isempty(kwargs) && error("Composition of parametric gates not allowed")
-    gate1 = name[1:prevind(name, pluspos.start)]
-    gate2 = name[nextind(name, pluspos.start):end]
+    ignore() do 
+      gate1 = name[1:prevind(name, pluspos.start)]
+      gate2 = name[nextind(name, pluspos.start):end]
```

Comment:
Once `@non_differentiable ITensors.name(::Any)` is defined, hopefully the `ignore()` here will become unnecessary.

Reply:
This however I seem not to be able to fix using the macro. Even by adding `@non_differentiable prevind` and `@non_differentiable nextind`